### PR TITLE
Write comments with verifiable examples

### PR DESCRIPTION
**Pull request by lzrski**
Some functions and decoders are covered by doc-tests (examples with expected return values). The tests are performed as part of Make targets: verify-examples and tests.

Some more complex cases (like viewSalary with all data provided) were intentionally ommited. The examples are primarily for human readers, so they shouldn't be too complex. If we need to test these complex scenarios, we should use unit or integration tests.

**Notes**

This PR supersedes #22 

Unfortunately the doc tests currently do not get counted in coverage report. This is due to how Elm Validate Examples work. To change this first that issue would have to be addressed: https://github.com/stoeffel/elm-verify-examples/issues/65

Another improvement would be to follow the advise here https://github.com/elm-explorations/test/tree/1.2.1#application-specific-techniques and not expose all the internal helpers from Main module. We could create several specialized modules like `Salary`, `City`, `Tenure` or just one `Internal` with everything exposed and one `Main` which would only expose `main`. IMO this goes beyond the scope of this PR. Maybe we should make an issue for that?

### CODE REVIEW

**Comment by zupo** [comment]
> Another improvement would be to follow the advise here https://github.com/elm-explorations/test/tree/1.2.1#application-specific-techniques and not expose all the internal helpers from Main module. We could create several specialized modules like Salary, City, Tenure or just one Internal with everything exposed and one Main which would only expose main. IMO this goes beyond the scope of this PR. Maybe we should make an issue for that?

Yes.

**Comment by zupo** [comment]
> Unfortunately the doc tests currently do not get counted in coverage report. This is due to how Elm Validate Examples work. To change this first that issue would have to be addressed: stoeffel/elm-verify-examples#65

Makes sense. Please work on getting https://github.com/stoeffel/elm-verify-examples/issues/65 resolved, merged & release.

File: src/SalaryCalculator.elm [hunk 1]
```diff
@@ -42,6 +56,21 @@ import Url.Parser.Query as QueryParser exposing (int, map3, string)
 -- MODEL
 
 
+{-| Given a list of records with a name field returns first item with a matching name. If no item has a matching name then returns Nothing.
```
Comment (zupo) [inline src/SalaryCalculator.elm]: 80 char limit please (with 10% leeway, so absolute max 88).

Comment (lzrski) [reply src/SalaryCalculator.elm]: Done.

**Comment by tad-lispy** [comment]
> Please work on getting stoeffel/elm-verify-examples#65 resolved, merged & release.

My PR finally got merged and released! https://github.com/stoeffel/elm-verify-examples/pull/84

**Pull request merged**